src/main/java/com/google/firebase/auth/FirebaseAuth.java

```diff
@@ -114,6 +111,108 @@ public static synchronized FirebaseAuth getInstance(FirebaseApp app) {
+    checkNotNull(options, "options must not be null");
+    return call(new Callable<String>() {
+      @Override
+      public String call() throws Exception {
+        return userManager.createSessionCookie(idToken, options);
+      }
+    });
+  }
+
+  /**
+   * Creates a new Firebase session cookie from the given ID token and options. The returned JWT
+   * can be set as a server-side session cookie with a custom cookie policy.
+   *
+   * @param idToken The Firebase ID token to exchange for a session cookie.
+   * @param options Additional options required to create the cookie.
+   * @return An {@code ApiFuture} which will complete successfully with a session cookie string.
+   *     If an error occurs while generating the cookie or if the specified ID token is invalid,
+   *     the future throws a {@link FirebaseAuthException}.
+   * @throws IllegalArgumentException If the ID token is null or empty, or if options is null.
+   */
+  public ApiFuture<String> createSessionCookieAsync(
+      @NonNull String idToken, @NonNull SessionCookieOptions options) {
+    return new TaskToApiFuture<>(createSessionCookie(idToken, options));
+  }
+
+  /**
+   * Parses and verifies a Firebase session cookie.
+   *
+   * <p>If the cookie is valid, the returned Future will complete successfully and provide a
+   * parsed version of the cookie from which the UID and other claims can be read.
+   * If the cookie is invalid, the future throws an exception indicating the failure.
+   *
+   * <p>This does not check whether the cookie has been revoked. See
+   * {@link #verifySessionCookie(String, boolean)}.
+   *
+   * @param cookie A Firebase session cookie string to verify and parse.
+   * @return An {@code ApiFuture} which will complete successfully with the parsed cookie, or
+   *     unsuccessfully with the failure Exception.
+   */
+  public ApiFuture<FirebaseToken> verifySessionCookieAsync(String cookie) {
+    return new TaskToApiFuture<>(verifySessionCookie(cookie, false));
+  }
+
+  /**
+   * Parses and verifies a Firebase session cookie.
+   *
+   * <p>If the cookie is valid, the returned Future will complete successfully and provide a
+   * parsed version of the cookie from which the UID and other claims can be inspected.
+   * If the cookie is invalid, the future throws an exception indicating the failure.
+   *
+   * <p>If {@code checkRevoked} is true, additionally checks if the cookie has been revoked.
+   *
+   * <p>If the cookie is valid, and not revoked, the returned Future will complete successfully and
+   * provide a parsed version of the cookie from which the UID and other claims can be read.
+   * If the cookie is invalid or has been revoked, the future throws an exception indicating the
+   * failure.
+   *
+   * @param cookie A Firebase session cookie string to verify and parse.
+   * @param checkRevoked A boolean indicating whether to check if the cookie was revoked.
```

Comment:
Nit: Can you be more explicit and state that "revoked" and "expiry" are two different things? Even just using "manually revoked" would make this more clear.